protocol/src/traits/executor.rs

```diff
@@ -52,3 +54,15 @@ pub trait Executor {
         request: &TransactionRequest,
     ) -> ProtocolResult<ExecResp>;
 }
+
+pub trait Dispatcher {
+    fn call(&self, context: ServiceContext, readonly: bool) -> ProtocolResult<ExecResp>;
+}
+
+pub struct DispatcherHolder;
+
+impl Dispatcher for DispatcherHolder {
```

Comment:
More reasonable to name it `NoopDispatcher`.

Reply:
renamed